io/src/csv/read_cells_alt.cpp

```diff
@@ -0,0 +1,169 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "read_cells_alt.hpp"
+
+#include "make_cell_reader.hpp"
+
+// System include(s).
+#include <algorithm>
+#include <cassert>
+#include <utility>
+#include <vector>
+
+namespace {
+
+/// Comparator used for sorting cells. This sorting is one of the assumptions
+/// made in the clusterization algorithm
+const auto comp = [](const traccc::alt_cell& c1, const traccc::alt_cell& c2) {
+    return c1.c.channel1 < c2.c.channel1;
+};
```

Comment:
Is there a good reason for this to be declared using anonymous function syntax? Why not just have a straight-up function?

Reply:
I thought this was the "cleanest" way of doing defining a custom comparator to call in std::sort but maybe something to do differently in the future?

Reply:
In principle I would consider this to be a less clean solution which might also have more overhead than simply using a function pointer. That being said the compiler should be more than able to defunctionalize this.

I hope.

Reply:
Either way let's not worry about it now.